iscsidev/iscsi.go

```diff
@@ -132,21 +131,10 @@ func (dev *Device) StartInitator() error {
 	if err := iscsi.LoginTarget(localIP, dev.Target, ne); err != nil {
 		return err
 	}
-	if dev.Device, err = iscsi.GetDevice(localIP, dev.Target, TargetLunID, ne); err != nil {
+	if dev.KernelDevice, err = iscsi.GetDevice(localIP, dev.Target, TargetLunID, ne); err != nil {
 		return err
 	}
 
-	deviceFound := false
-	for i := 0; i < RetryCounts; i++ {
```

Comment:
Previously the device name discovery and device discovery are two steps. We can merge them into one but we need to make sure we will wait long enough for that.

Reply:
Since we no longer rely on the creation of a device node and the getScsciDevice now returns the KernelDevice which it evaluates via lsblk we know for a fact that the device is available, so no need for the additional checking. Since we already did it as part of the getScsciDevice call.

Reply:
We might need to increase the rety count for the getScsi call, but the moment it returns sdba the device will always be available since it gets that name from the kernel.

Reply:
Make sense. Though the previous retry is only 1 second * 5, so it's better to increase the timeout a bit. Otherwise, it may fail in some unexpected high load environment.

Reply:
We certainly can increase the retry counts but it should not be necessary.
Since even in the prior code we would have failed early if iscsi.GetDevice did not return a device name.

It seems like the only reason we had the second retry loop is that we where dependent on the creation of the device node. While now we no longer have that dependency we only need to know that the kernel has the device registered.
Which it should once isci returns the device name, since that's were it comes from?

Reply:
Make sense. I was being more cautious since we cannot change the timeout after shipping the release. If we get it wrong here, the only way to fix it is to ship a new release and ask all the users to upgrade. But if iscsiadm knows it's e.g. `sdb`, then `sdb` should has been created in the kernel already. If that's the case, the extra caution here is not needed.

I will leave this to you to decide if the timeout needs to be extended.

Reply:
I don't think the timeout needs to be increased, but I will go a head and increase the retryCount inside of the iscsi.GetDevice function since we freed up 15s of time by removing the extra check loop.

We can increase that checkCount withhout increasing our totalTime taken for the same codepath.
I doubled the retryCount from 5 to 10 which adds an additional 5s window for success.